server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteIndexRecoveryIT.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.remotestore;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.index.IndexModule;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.indices.recovery.IndexRecoveryIT;
+import org.opensearch.indices.replication.common.ReplicationType;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.nio.file.Path;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class RemoteIndexRecoveryIT extends IndexRecoveryIT {
+
+    protected static final String REPOSITORY_NAME = "test-remore-store-repo";
+
+    protected Path absolutePath;
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.REMOTE_STORE, "true").build();
+    }
+
+    @Override
+    protected void afterFirstStartNode() {
```

Comment:
lets move  this to

```
    @Before
    public void setup() {
        setupRepo();
    }
```

Reply:
In the parent IT (i.e. IndexRecoveryIT), the data node is getting created in the test method itself. I did try adding a cluster manager in the setup method, but there are certain tests that fail where there are assertions of node count. In interest of keeping the existing test intact, I have added this method which registers the repository required for creating remote enabled index.

Reply:
The problem with this approach is that any new test need we add, we need to remember to call `setupRepo`. IMO, we should fix the tests that are failing on the node count. Instead of all the nodes in the cluster, these tests can check only data nodes.

Reply:
Have made the fix as suggested.